### DISCUSSION TITLE
Installing NextJS app in cPanel Application Manager

### DISCUSSION THREAD

**Discussion by heylols** [vercel/next.js#31372] [Help]
I have followed this tutorial to setup a NodeJS inside the cPanel : https://docs.cpanel.net/knowledge-base/web-services/how-to-install-a-node.js-application/

And it expects app.js file as the entry point. As NextJS doesn't have one by default, I checked the documentation and found about the Custom Server : https://nextjs.org/docs/advanced-features/custom-server

But it says using Custom Server removes performance optimizations. Is there any way to keep the performance optimizations? Because performance is something critical for this app.

Btw the server is a Dedicated server of my client and they have many other websites running and wants to use that server for this app.

Thanks

**Comment by schizek** [comment]
I'm looking for the answer to this as well.  Is a [Custom Server](https://nextjs.org/docs/advanced-features/custom-server) the only option for running a Next.js application in cPanel?
